cmd/aem/osgi.go

```diff
@@ -679,10 +681,18 @@ func osgiConfigDefineFlags(cmd *cobra.Command) {
 
 func osgiConfigFromFlag(cmd *cobra.Command, i pkg.Instance) *pkg.OSGiConfig {
 	pid, _ := cmd.Flags().GetString("pid")
-	config := i.OSGI().ConfigManager().ByPID(pid)
+	var config pkg.OSGiConfig
+	if isFactoryPID(pid) {
+		config = i.OSGI().ConfigManager().ByFactoryPID(pid)
+	} else {
+		config = i.OSGI().ConfigManager().ByPID(pid)
+	}
 	return &config
 }
 
+func isFactoryPID(pid string) bool {
```

Comment:
methods are not scoped to files; they are package wide; so... maybe..

`isOsgiConfigFactoryPid()`